export/client/clients/couch-client.go

```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
+ * Copyright 1995-2018 Hitachi Vantara Corporation. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+package clients
+
+import (
+	"strconv"
+	"github.com/flimzy/kivik"
+	"context"
+	_ "github.com/go-kivik/couchdb"
+	"github.com/edgexfoundry/edgex-go/export"
+	"gopkg.in/mgo.v2/bson"
+	"github.com/edgexfoundry/edgex-go/pkg/models"
+	"time"
+)
+
+type CouchClient struct {
+	Database *kivik.DB
+	Timeout  time.Duration
+}
+
+type CouchRegistration struct {
+	ID          bson.ObjectId            `bson:"_id,omitempty" json:"_id,omitempty"`
+	Rev         string                   `json:"_rev,omitempty"` //required for update
+	Created     int64                    `json:"created"`
+	Modified    int64                    `json:"modified"`
+	Origin      int64                    `json:"origin"`
+	Name        string                   `json:"name,omitempty"`
+	Addressable models.Addressable       `json:"addressable,omitempty"`
+	Format      string                   `json:"format,omitempty"`
+	Filter      export.Filter            `json:"filter,omitempty"`
+	Encryption  export.EncryptionDetails `json:"encryption,omitempty"`
+	Compression string                   `json:"compression,omitempty"`
+	Enable      bool                     `json:"enable"`
+	Destination string                   `json:"destination,omitempty"`
+}
+
+// Return a pointer to the CouchClient
+func newCouchClient(config DBConfiguration) (*CouchClient, error) {
+	// Create the dial info for the Couch session
+
+	connectionString := "http://" + config.Username + ":" + config.Password + "@" + config.Host + ":" + strconv.Itoa(config.Port)
+
+	timeout := time.Duration(config.Timeout) * time.Millisecond
+
+	ctx1, cancel1 := context.WithTimeout(context.Background(), timeout)
+	defer cancel1()
+	client, err := kivik.New(ctx1, "couch", connectionString)
```

Comment:
`ctx`es are used only within `kivik.New()`, so you can reuse same `ctx, cancel` variables instead declaring new ones each time.